app/drivers/zephyr/kscan_gpio_matrix.c

```diff
@@ -181,19 +181,19 @@ static int kscan_gpio_config_interrupts(struct device **devices,
         struct kscan_gpio_data_##n *data = CONTAINER_OF(work, struct kscan_gpio_data_##n, work);   \
         kscan_gpio_read_##n(data->dev);                                                            \
     }                                                                                              \
-    COND_CODE_1(CONFIG_ZMK_KSCAN_MATRIX_POLLING, (),                                               \
```

Comment:
Removed the conditional compilation on `kscan_gpio_irq_callback_handler_##n`. Polling mode calls this function as well.